app/server.rb

```diff
@@ -0,0 +1,62 @@
+require 'sinatra/base'
+require 'rack-flash'
+require_relative 'data_mapper_setup'
+require_relative './models/user'
+require_relative './session_helper'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  set :session_secret, 'superbly secret'
+  use Rack::Flash
+  use Rack::MethodOverride
+  include SessionHelper
+
+  get '/' do
+    erb :index
+  end
+
+  post '/users' do
+    user = User.new(email: params[:email],
+                    name: params[:name],
+                    password: params[:password],
+                    username: params[:username])
+    if user.save
+      create_session(user)
+    else
+      flash.now[:errors] = user.errors
+    end
+    erb :index
+  end
+
+  get '/users/:id' do
+    @user = User.first(id: params[:id])
+    @users_peeps = @user.peeps.all(order: [:id.desc])
```

Comment:
'Good thin controllers' should really only have one instance variable to render. They can access parts of that instance variable, but otherwise you end with a Frankenstinian monster of instance variables everywhere.

You can access `@user` in your template to iterate over.